src/components/namestable/worker-edit/worker-name-edit-field.components.tsx

```diff
@@ -0,0 +1,79 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at https://mozilla.org/MPL/2.0/. */
+import { Grid, TextField, Typography } from "@material-ui/core";
+import React, { useCallback, useEffect } from "react";
+import { useSelector } from "react-redux";
+import { ApplicationStateModel } from "../../../state/models/application-state.model";
+import { FormFieldErrorLabelStack } from "./form-field-error-label.component";
+import {
+  FormFieldOptions,
+  useFormFieldStyles,
+  WorkerEditComponentMode,
+} from "./worker-edit.models";
+
+export interface WorkerNameEditFieldOptions extends FormFieldOptions {
+  workerName: string;
+  setWorkerName: (newWorkerName: string) => void;
+  mode: WorkerEditComponentMode;
+}
+
+export function WorkerNameEditField({
+  workerName,
+  setWorkerName,
+  setIsFieldValid,
+  mode,
+}: WorkerNameEditFieldOptions): JSX.Element {
+  const classes = useFormFieldStyles();
+
+  const workerNames = useSelector((state: ApplicationStateModel) =>
+    Object.keys(state.actualState.persistentSchedule.present.employee_info.type)
+  );
+
+  const isWorkerWithSameNameExists = useCallback((): boolean => {
+    const isWorkerNameInvalid =
```

Comment:
myślę że ta zmienna nie jest potrzebna

Reply:
Preferuję wynosić wyniki operacji do zmiennych nawet jeśli ta zmienna będzie w następnej linii zwrócona w returnie. 
Główna zaleta tego - to  łatwiejsze debugowanie, bo czasami chcesz zobaczyć co zwraca funkcja będą w tej funkcji i wtedy taka zmienna na to pozwala.